services/ens/api.go

```diff
@@ -0,0 +1,312 @@
+package ens
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multibase"
+	"github.com/multiformats/go-multihash"
+	"github.com/pkg/errors"
+	"github.com/status-im/status-go/rpc"
+	"github.com/wealdtech/go-multicodec"
+)
+
+func NewAPI(rpcClient *rpc.Client) *API {
+	return &API{
+		contractMaker: &contractMaker{
+			rpcClient: rpcClient,
+		},
+	}
+}
+
+type API struct {
+	contractMaker *contractMaker
+}
+
+func (api *API) Resolver(ctx context.Context, chainID uint64, username string) (*common.Address, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return nil, err
+	}
+
+	registry, err := api.contractMaker.newRegistry(chainID)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	resolver, err := registry.Resolver(callOpts, nameHash(username))
+	if err != nil {
+		return nil, err
+	}
+
+	return &resolver, nil
+}
+
+func (api *API) OwnerOf(ctx context.Context, chainID uint64, username string) (*common.Address, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return nil, err
+	}
+
+	registry, err := api.contractMaker.newRegistry(chainID)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	owner, err := registry.Owner(callOpts, nameHash(username))
+	if err != nil {
+		return nil, nil
+	}
+
+	return &owner, nil
+}
+
+func (api *API) ContentHash(ctx context.Context, chainID uint64, username string) ([]byte, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return nil, err
+	}
+
+	resolverAddress, err := api.Resolver(ctx, chainID, username)
+	if err != nil {
+		return nil, err
+	}
+
+	resolver, err := api.contractMaker.newPublicResolver(chainID, resolverAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	contentHash, err := resolver.Contenthash(callOpts, nameHash(username))
+	if err != nil {
+		return nil, nil
+	}
+
+	return contentHash, nil
+}
+
+func (api *API) PublicKeyOf(ctx context.Context, chainID uint64, username string) ([32]byte, [32]byte, error) {
```

Comment:
I wonder if it would be a good idea to instead of returning X and Y, to return instead a `ecdsa.PublicKey`?